libevmasm/ConstantOptimiser.h

```diff
@@ -100,7 +103,7 @@ class LiteralMethod: public ConstantOptimisationMethod
 	explicit LiteralMethod(Params const& _params, u256 const& _value):
 		ConstantOptimisationMethod(_params, _value) {}
 	virtual bigint gasNeeded() override;
```

Comment:
It returns a combination of runtime and deploy costs of the actual routine. As we also have a "LiteralMethod", we compare it also to the "current costs". The combination is done by a crude rade-off analysis: `_params` contains a parameter called `runs` which is the assumed number this piece of the code is executed after deployment.